댓글 수정, 삭제 관련 PR 요청 드립니다!

leejengjun opened the pull request:
기존 기능에서 수정한 부분 입니다.
- 게시물 기간 검색 코드 수정.
- 부트스트랩 관련 파일 

새롭게 추가한 부분 입니다.
- 댓글 수정 기능
- 댓글 삭제 기능

leejengjun committed:
Merge branch 'leejengjun-study-board#12'

leejengjun committed:
Merge branch 'master' of https://github.com/zerom-H/giantstep-study-board

leejengjun committed:
Refactor : 수정 게시물 기간 검색 코드

- 수정
  - 최초 페이지 로드 시 기간에 기본값 설정.
    - 뷰 파일 코드에 script 태그 위치 변경.
  - BoardRepositoryImpl 에 기간 검색 로직을 between -> goe , loe 로 변경.
    - 이에 따른 로직 수정.
  - 게시물 검색 테스트 코드 수정.

Resolve: giantstep-study-board #8

leejengjun committed:
Feat : 추가 부트스트랩 관련 파일

- 부트스트랩 관련 파일들 추가.

Resolve: giantstep-study-board #15

leejengjun committed:
Feat : 추가 댓글 수정 기능

- boardOneDetail.html
  - 댓글 수정 폼 추가 및 자바스크립트로 유효성 검사 구현.
- 댓글 수정 로직 구현
  - 댓글 수정은 변경감지를 활용.
  - 댓글 수정폼Dto 생성.
- BoardCommentAddFormDto 필드명 수정.
- BoardCommentListDto 필드에 boardCommentPassword 추가.
- BoardCommentRepositoryImpl 댓글 저체조회 메소드에 boardComment.password 추가.

Resolve: giantstep-study-board #13

leejengjun committed:
Feat : 추가 댓글 삭제 기능

- boardOneDetail.html
  - 댓글 삭제 폼(모달 활용) 추가 및 자바스크립트로 유효성 검사 구현.
- 댓글 삭제 로직 구현
  - 삭제 기능은 삭제여부를 확인할 수 있는 컬럼을 업데이트 하는 소프트 방식 적용.
    - 이에 따른 아래 클래스에 deletedYn 필드 관련하여 수정
      - BoardComment
      - BoardCommentAddFormDto
      - BoardCommentRepositoryImpl
  - 댓글 삭제는 변경감지를 활용.
- 댓글 삭제 테스트 코드
  - 기존 단위 테스트에서 given 부분 BoardComment 객체 생성 시 필드에 deletedYn 부분 추가.

Resolve: giantstep-study-board #14

leejengjun committed:
Refactor : 수정 댓글 수정 관련 코드

- boardOneDetail.html
  - 댓글 id를 넘기는 것이 아닌 DTO 내에 해당 필드를 추가해서 DTO 로 넘기게 수정.
- BoardCommentController commentUpdateDone 메소드 부분 sout 코드 삭제.
- BoardCommentService updateBoardComment 메소드 부분 매개변수 boardCommentId 삭제.
- BoardCommentServiceTest 댓글_수정 단위 테스트 부분 수정.

Resolve: giantstep-study-board #13

leejengjun committed:
Refactor : 수정 댓글 삭제 비밀번호 체크 폼

- boardOneDetail.html
  - 댓글 삭제 클릭시 보이는 모달 창에 댓글 번호와 댓글 비밀번호 hidden 처리.

Resolve: giantstep-study-board #14

JunSubHyun commented on src/main/java/com/giantstep/board/domain/board/dto/comment/BoardCommentListDto.java:
```diff
@@ -12,15 +12,17 @@ public class BoardCommentListDto {
         this.boardCommentWriter = boardCommentWriter;
         this.boardCommentContents = boardCommentContents;
         this.boardId = boardId;
+        this.boardCommentPassword = boardCommentPassword;
```
개인적인 의견으로는 댓글에는 비밀번호가 필요없다는 의견이 있습니다.
비밀번호까지 입력하게 되면 게시글이랑 차이가 없어서 가볍게 리플다는 댓글에 대한 이점이 사라지는것 같습니다

JunSubHyun replied:
밑에 보다보니 로그인이 없는걸 까먹었습니다.
세션없이 구현하다보면 비밀번호가 나을꺼같습니다.

JunSubHyun commented on src/main/resources/templates/board/boardOneDetail.html:
```diff
@@ -76,12 +77,55 @@ <h2 th:text="#{label.board.oneDetailTitle}">글 상세보기</h2>
+                            <div class="modal-body">
+                                <form id="commentDeleteCheckForm">
+                                    <input hidden id="deleteBoardCommentId" th:value="${boardCommentList.boardCommentId}">
+                                    <input hidden th:value="${boardCommentList.boardCommentPassword}">
```
이렇게 되면 비밀번호가 쉽게 노출이 될꺼예요 (개발도구만 켜도 보일거에요)
비밀번호를 처음에 같이 불러서 비동기로 비교하는 방식은 비밀번호 노출이 너무 쉬워서 비밀번호 역할이 제대로 안될꺼 같습니다.

JunSubHyun commented on src/main/resources/templates/board/boardOneDetail.html:
```diff
@@ -76,12 +77,55 @@ <h2 th:text="#{label.board.oneDetailTitle}">글 상세보기</h2>
+                            </div>
+                            <div class="modal-body">
+                                <form id="commentDeleteCheckForm">
+                                    <input hidden id="deleteBoardCommentId" th:value="${boardCommentList.boardCommentId}">
```
조금 옛날 생각일수도 있는데 HTML 에서 ID 값은 중복이 되면 안되는걸로 알고 있습니다.
그리고 Javascript 실행 하실때 Id 값으로 핸들링 하시기 때문에 id 값이 중복이 되면
문제를 야기 할 수 있습니다.

leejengjun committed:
Refactor : 수정 준섭님 코멘트(댓글 삭제 비밀번호 체크 폼)

- boardOneDetail.html
  - 뷰단에 게시글 댓글 삭제 폼에서 password 노출 제거(hidden 처리된 input 태그 제거)
  - ajax 를 활용해 비동기식으로 댓글 삭제 체크 로직 수정.

- 뷰단 로직 수정에 따른 컨트롤러 서비스 리포지토리 수정.
  - BoardCommentController
    - commentDeleteDone 메소드 수정.
  - BoardCommentService
    - deleteBoardComment 해당 로직 수정.
      - AtomicReference , Optional<T>, ifPresent() 활용.
  - BoardCommentRepository
    - findBoardCommentByIdAndPassword 쿼리 메소드 추가.
  - 댓글 리스트 DTO 에 password 필드 제거.
    - BoardCommentListDto
    - BoardCommentRepositoryImpl

Resolve: giantstep-study-board #14

leejengjun committed:
Feat : 추가 댓글 단건 조회 테스트 코드

- BoardCommentServiceTest
  - findBoardCommentByIdAndPassword() 에 대한 테스트 코드 작성

Resolve: giantstep-study-board #14

leejengjun committed:
Refactor : 수정 댓글 수정 폼

- boardOneDetail.html
  - 최초 콜렙스 -> 모달로 변경
  - 뷰단에 hidden input 태그로 가져온 비밀번호 노출 제거.
  - 비밀번호 검증을 뷰단에서 하는게 아닌 ajax 를 활용해서 백단에서 검증 후 처리하도록 수정.
- BoardComment
  - 엔티티 내부 메소드 updateBoardComment 메소드의 리턴 타입을 BoardComment -> Boolean 으로 변경.
- BoardCommentController
  - commentUpdateDone 메소드 수정.
- BoardCommentService
  - updateBoardComment 메소드 로직 수정.
- BoardCommentUpdateFormDto
  - updateBoardCommentID 필드명 수정.

Resolve: giantstep-study-board #13

leejengjun committed:
Refactor : 수정 댓글 수정, 댓글 삭제 폼 버그

- 이 커밋 바로 앞까지해서 생각지 못한 버그가 발견되었습니다.
- 결과적으로 버그의 원인은 준섭님이 코멘트 달아주신 HTML 에서 'id 중복'으로인해 발생한 버그였습니다.
- 이 버그는 th:onclick() 에 매개변수를 전달해줌으로서 해결하였습니다.
- 준섭님의 코멘트 해주신 부분대로 뷰단에 불필요한 정보(비밀번호) 노출을 제거하였습니다.

- boardOneDetail.html
  - 댓글 수정 모달, 댓글 삭제 관련 코드 수정.
  - 불필요한 태그 코드 정리.
- BoardCommentController
  - commentUpdateDone() 메소드 수정.
  - commentDeleteDone() 메소드 수정.
- BoardCommentDeleteCheckRequest 추가.
  - 뷰에서 JSON 으로 전달하는 댓글 삭제에 필요한 데이터를 받는 DTO
- BoardCommentServiceTest
  - 댓글 수정 단위 테스트 내용 수정.
- BoardCommentUpdateFormDto 클래스 명 수정.
  - BoardCommentUpdateCheckRequest 로 명명.

Resolve: giantstep-study-board #13

leejengjun committed:
Refactor : 수정 BoardController

- 클래스 내 boardOneDetailView() 메소드 파라미터 BoardCommentUpdateFormDto 이름 변경.
  - BoardCommentUpdateCheckRequest 로 명명.

Resolve: giantstep-study-board #4
